File: javascript/twoStrings.js
function twoStrings(s1, s2) {
  const charHash = {};
  s1.split("").forEach((letter, i) => {
    charHash[letter] = charHash[letter] + 1 || 1;
  });

  let match = "NO";
  s2.split("").forEach(letter => {
    if (charHash[letter] >= 1) match = "YES";
  });
  return match;
}

s1 = "hackerrankcommunity";
s2 = "cdecdecdecde";
// s1 = "wouldyoulikefries";
// s2 = "abcabcabcabcabcabc";
console.log(twoStrings(s1, s2));

File: ruby/two_strings.rb
# Complete the twoStrings function below.
def twoStrings(s1, s2)
  s_hash = Hash.new(0)
  s1.split('').each do | el |
      s_hash[el] += 1
  end

  found = 'NO'
  s2.split('').each do |el|
    found = "YES" if s_hash[el] > 1
  end

  found
end

twoStrings('apple', "hi")